src/TorchSharp/TorchVision/dsets/MNISTReader.cs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) .NET Foundation and Contributors.  All Rights Reserved.  See LICENSE in the project root for license information.
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using TorchSharp.torchvision;
+using static TorchSharp.torch;
+using static TorchSharp.torch.utils.data;
+
+namespace TorchSharp.torchvision.dsets
+{
+    /// <summary>
+    /// Data reader utility for datasets that follow the MNIST data set's layout:
+    ///
+    /// A number of single-channel (grayscale) images are laid out in a flat file with four 32-bit integers at the head.
+    /// The format is documented at the bottom of the page at: http://yann.lecun.com/exdb/mnist/
+    /// </summary>
+    public sealed class MNISTReader : Dataset
+    {
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="path">Path to the folder containing the image files.</param>
+        /// <param name="prefix">The file name prefix, either 'train' or 't10k' (the latter being the test data set).</param>
+        /// <param name="device">The device, i.e. CPU or GPU to place the output tensors on.</param>
+        /// <param name="transform"></param>
+        public MNISTReader(string path, string prefix, Device device = null, ITransform transform = null)
```

Comment:
'device' is not used in this constructor -- it needs to be used to indicate where to place the tensors that are created.